app/src/main/java/com/practicum/playlistmaker/new_playlist/ui/fragment/NewPlaylistFragment.kt

```diff
@@ -0,0 +1,264 @@
+                buttonCreateNewPlaylist.isEnabled = true
+                namePlaylist.setBoxStrokeColorStateList(
+                    resources.getColorStateList(
+                        R.color.text_input_box_stroke,
+                        null
+                    )
+                )
+                namePlaylist.defaultHintTextColor =
+                    ContextCompat.getColorStateList(requireContext(), R.color.text_input)
+            }
+
+            override fun afterTextChanged(s: Editable?) {
+                if (namePlaylistText.text.isNullOrEmpty()) {
+                    buttonCreateNewPlaylist.isEnabled = false
+                    namePlaylist.setBoxStrokeColorStateList(
+                        resources.getColorStateList(
+                            R.color.text_not_box_stroke,
+                            null
+                        )
+                    )
+
+                    namePlaylist.defaultHintTextColor =
+                        ContextCompat.getColorStateList(requireContext(), R.color.text_not)
+                }
+            }
+        })
+
+        descriptionPlaylistText.addTextChangedListener(object : TextWatcher {
+            override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {}
+
+            override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                descriptionPlaylist.setBoxStrokeColorStateList(
+                    resources.getColorStateList(
+                        R.color.text_input_box_stroke,
+                        null
+                    )
+                )
+                descriptionPlaylist.defaultHintTextColor =
+                    resources.getColorStateList(R.color.text_input, null)
+            }
+
+            override fun afterTextChanged(s: Editable?) {
+                if (namePlaylistText.text.isNullOrEmpty()) {
+                    descriptionPlaylist.setBoxStrokeColorStateList(
+                        resources.getColorStateList(
+                            R.color.text_not_box_stroke,
+                            null
+                        )
+                    )
+                    descriptionPlaylist.defaultHintTextColor =
+                        resources.getColorStateList(R.color.text_not, null)
+                }
+            }
+        })
+
+        coverPlaylist.setOnClickListener() {
+            requestPermission()
+        }
+
+        buttonCreateNewPlaylist.setOnClickListener() {
+
+            if (oldCoverDrawable != coverPlaylist.drawable) {
+                newPlaylistViewModel.saveImage(namePlaylistText.text.toString())
+            }
+
+            newPlaylistViewModel.createPlaylistClicked(
+                namePlaylistText.text.toString(), descriptionPlaylistText.text.toString()
+            )
+
+            showMessageCreatePlaylist()
+
+            exit()
+        }
+    }
+
+    private fun exit() {
+        findNavController().navigateUp()
```

Comment:
🍏 Возможно, здесь было бы достаточно использовать стандартный подход:
```suggestion
        findNavController().popBackStack()
```